### DISCUSSION TITLE
Fishing system

### DISCUSSION THREAD

**Discussion by PackJC** [EveronLife/EveronLife#82] [Ideas]
**(reference my dayz fishing mod)**
https://github.com/PackJC/gebsfish

Fish are caught all throughout the map, where ever water is found. 

- Fresh & Salt Water Fish
- Fishing License (Must have to sell legal fish to the trader) (only given ticket or wanted level increased if police cite you, no automatic wanted level for fishing without)
- Illegal fish, use "poaching fishing rod" to catch illegal fish. Illegal fish do not require license to sell. Poachers trader similiar to drug trader in unrestricted zone.
- Weighted chanced-based system for the fish (30% chance to catch carp etc...)
- Fishing Pole with durability (Bamboo pole? can be low poly and still look good) Fishing pole works in both waters.
- Fishing Action: Press F when pole is in hand and looking at water. Chance of catching fish. Pole slow degrade over time. If fish caught, fish will appear at feet of player. Fishing pole can be repaired with "fishing line" which could be bought at trader. Having repair option for the pole will enable a buff for fishing, deters people from buying a bunch of fishing poles. Instead keeping one, repairing it and giving an RP experience. 
- Large fish in hand will make player slow walk (such as bluefin tuna)
- Largest fish cannot be put into hands, only dragged into vehicle inventory (such as shark)

**Purposed Saltwater Fish** (needs to be common in north sea area)
- Minnow
- Carp
- Trout

**Purposed Freshwater Fish** (needs to be common in north sea area)
- Mackerel
- Bluefin Tuna
- Atlantic Cod

**Purposed Illegal Fish** (needs to be common in north sea area)
- Sharks

The goal for the fish models are to be size efficient as possible so it can work on consoles and pc. This means low poly, simple models - we just need to find a good spot inbetween. 

**Minnow (fresh water)**
https://github.com/PackJC/EveronLife/tree/main/src/Assets/Items/Fish/Minnow
![image](https://user-images.githubusercontent.com/34726562/173877368-91b11b70-deec-4e07-adaa-8f5b1eac551a.png)

**Trout (fresh water)**
https://github.com/PackJC/EveronLife/tree/main/src/Assets/Items/Fish/Trout
![image](https://user-images.githubusercontent.com/34726562/173948700-e9d88128-cc09-4e85-aeac-04cf28129384.png)

**Carp (fresh water)**
https://github.com/PackJC/EveronLife/tree/main/src/Assets/Items/Fish/Carp
![image](https://user-images.githubusercontent.com/34726562/173976939-bc69305b-c0e5-4ada-96a9-1ba2b9715616.png)

**Pacific Mackerel (salt water)**
https://github.com/PackJC/EveronLife/tree/main/src/Assets/Items/Fish/Mackerel
![image](https://user-images.githubusercontent.com/34726562/173877431-f88f7e91-4588-499f-a8f2-555738552ab1.png)

**Bluefin Tuna (salt water)**
https://github.com/PackJC/EveronLife/tree/main/src/Assets/Items/Fish/Tuna
![image](https://user-images.githubusercontent.com/34726562/173960864-fcaf7a5e-9b9a-4a07-9095-a29acd8dc675.png)

**Atlantic Cod (salt water)**
https://github.com/PackJC/EveronLife/tree/main/src/Assets/Items/Fish/Cod
![image](https://user-images.githubusercontent.com/34726562/173981866-49cd2bb7-d0af-434e-9e54-04720360388f.png)

**Shark (salt water & ILLEGAL) **
https://github.com/PackJC/EveronLife/tree/main/src/Assets/Items/Fish/Shark
![image](https://user-images.githubusercontent.com/34726562/174002646-60a1bf79-312b-4dee-a631-baa11e9bab07.png)

**Comment by thebonbon** [comment]
Ohh I didnt even think about fishing. We need this 100%.

**Reply by PackJC** [reply]
I'm not sure how quick it would be able to port to Reforger, but a ton of the work is already done on DayZ. (Which of course could be improved upon)

**Comment by SaxW** [comment]
Howdy, thought it worth mentioning here in case it may be easier to link it to fishing than mining. Had an idea that rather than mining for gold, you could pan for gold - ie
![image](https://user-images.githubusercontent.com/53883603/171979050-d8d9f11d-238e-46be-bf1f-3c11dcf1a12c.png)

There are a few rivers on the map that fit the bill
![image](https://user-images.githubusercontent.com/53883603/171979434-2be3d433-f388-49fa-a772-c1817af5ec7d.png)
![image](https://user-images.githubusercontent.com/53883603/171979453-56dc9817-e34a-428e-8305-f9d346dcc1bb.png)

**Reply by MarvinVomberg** [reply]
Sounds like a nice idea but isn't fully related to fishing. 
Could you please open an own discussion for gold panning?

**Comment by DrLuk** [comment]
With crab and clam fishing or something we could utilize scuba gear and boats. Maybe a big trawler for fishing out in the sea.

**Comment by 12gabriel3** [comment]
There are some assets in game, maybe we could use them somehow? 
Lobster traps, fishing nets and fishing net hangers
![image](https://user-images.githubusercontent.com/16885097/173883906-60f56d90-549b-4e5e-b695-0e5bcdeb89bd.png)

**Reply by PackJC** [reply]
Wow I did not see this at all. Incredibly useful and we should use them. Great props for around the trader too, like the fishing net piles

**Comment by 12gabriel3** [comment]
I think sharks, not only because of their size, but also because they are going to be a separate activity, should be of higher quality than regular assets, with increased poly count and a better resolution, id say about 500kb to 1 MB worth of textures and models. I also think the shark hunting activity should be like this:

Sharks spawn in a given area and swim around. The player has to go to that location, look for one and shoot it. He then proceeds to extract the fin and then sells it, as opposed to the whole shark.

**Reply by WhiskeyxD** [reply]
yea just the fin would make sense

**Comment by Pyth3rEx** [comment]
Just wanted to mention a few things:

- Why would illegal fish need an illegal fishing rod? Why not have one single rod, that can be bought either legally with a license to a trader or illegally to a rebel trader. those could also be stolen/looted by players.
- I think all fish should be both legal and illegal (other than a few select ones that are only illegal). eg. if I capture a tuna (which is officialy legal) but I lack a license, I cannot sell it to the official trader, and I can be fined. But maybe a black market trader could by it for a lower price/increased risks.
- Stopping people for buying multiple rods could be achieved by a big buy price for rods but a low maintenance price (eg. price for fishing line) making it more convenient to repair a rod than to buy multiple
- Fishing activity could also include an optional 'refinery' step, eg. going to a factory to get the fish skinned and sold for increased profit. Sharks could also have their dedicated mandatory factory making illegal shark fishing a bit more complicated but more cash output.
- Lastly.... we NEED turtle fishing.... #A3days

**Reply by WhiskeyxD** [reply]
100% bring back the turlas

**Comment by PackJC** [comment]
I have a turtle model, just haven't had time to put work into this system. I stopped putting effort into the game because of it's launch issues.
